lib/license_finder/license/text.rb

```diff
@@ -6,15 +6,28 @@ module Text
       SPACES = /\s+/.freeze
       QUOTES = /['`"]{1,2}/.freeze
       PLACEHOLDERS = /<[^<>]+>/.freeze
+      SPECIAL_SINGLE_QUOTES = /[‘’]/.freeze
+      SPECIAL_DOUBLE_QUOTES = /[“”„«»]/.freeze
+      ALPHABET_ORDERED_LIST = /\\\([a-z]\\\)\\\s/.freeze
+      ALPHABET_ORDERED_LIST_OPTIONAL = '(\([a-z]\)\s)?'
+      LIST_BULLETS = /(\d{1,2}\\\.|\\\*)\\\s/.freeze
+      LIST_BULLETS_OPTIONAL = '(\d{1,2}.|\*)\s*'
 
       def self.normalize_punctuation(text)
-        text.gsub(SPACES, ' ')
-            .gsub(QUOTES, '"')
-            .strip
+        text.dup.force_encoding('UTF-8')
+                 .gsub(SPECIAL_DOUBLE_QUOTES, '"')
+                 .gsub(SPECIAL_SINGLE_QUOTES, "'")
+                 .gsub(SPACES, ' ')
+                 .gsub(QUOTES, '"')
+                 .strip
       end
 
       def self.compile_to_regex(text)
-        Regexp.new(Regexp.escape(text).gsub(PLACEHOLDERS, '(.*)'))
+        Regexp.new(Regexp.escape(normalize_punctuation(text))
+                       .gsub(PLACEHOLDERS, '(.*)')
+                       .gsub(',', '(,)?')
```

Comment:
same here with new lines, we could either remove them or keep them optional.

```suggestion
                       .gsub(/\n+/, ' ')
                       .gsub(',', '(,)?')
```

Reply:
I have also recognized that some people have changed `COPYRIGHT HOLDER` to `COPYRIGHT OWNER`.

```ruby
.gsub('HOLDER', '(HOLDER|OWNER)')
```